app/services/order_cycle_form.rb

```diff
@@ -25,26 +25,50 @@ def save
 
   attr_accessor :order_cycle, :params, :permissions
 
-  def check_editable_schedule_ids
-    return unless params[:order_cycle][:schedule_ids]
-    requested = params[:order_cycle][:schedule_ids].map(&:to_i)
-    @existing_schedule_ids = @order_cycle.persisted? ? @order_cycle.schedule_ids : []
-    permitted = Schedule.where(id: requested | @existing_schedule_ids).merge(permissions.editable_schedules).pluck(:id)
-    result = @existing_schedule_ids
-    result |= (requested & permitted) # add any requested & permitted ids
-    result -= ((result & permitted) - requested) # remove any existing and permitted ids that were not specifically requested
+  def schedule_ids?
+    !!params[:order_cycle][:schedule_ids]
```

Comment:
Why not `params[:order_cycle][:schedule_ids].present?`?

Reply:
Is this just a stylistic preference? I don't particularly mind either way.

Reply:
Yes, I find double negative difficult to read/understand. Your version is pure Ruby though.